### DISCUSSION TITLE
Manually split files for critical css

### DISCUSSION THREAD

**Discussion by yellow1912** [tailwindlabs/tailwindcss#2699] [Ideas]
Hello all,

Even with purge css, the tailwind size can still be significantly big. My thinking is that we can manually decide what's important and should be included in above the fold file, and what should be included later.

For example, margin, padding, flex, grid rules (basically layout rules) should be used put in above the fold file. Something like color, animation, etc can be loaded later.

We can also run a post css critical css plugin to split file, but it can be challenging for dynamic websites with tons of different pages. The manual method can be a quicker way and yield good enough result.

**Comment by minimit** [comment]
Hello there's a big problem with splitting css into more css: **css specificity**.

Essentially it becomes a big problem the order in which you load the css, because as you know **css specified later has more specificity that css specified before**.

Tailwind has many components and utilities that needs to be loaded with the right order. For example if you add **any of your css** after any utility of tailwind, **they take precedence over the utility**.

Maybe It can be done in some way this splitting, but it's very sensitive and risky.

**Comment by MtDalPizzol** [comment]
This is something I'm also depending on.

I'm considering Tailwind for an old PHP project which is basically a news website and I'm following the [guide from web.dev on how to Defer non-critical CSS](https://web.dev/articles/defer-non-critical-css?hl=en).

But, having this process for each page on the website seems a complete overkill, since I'm running websites with tens of thousands of pages.

So, I'd like to do some splitting that I believe will be efficient enough for the majority of the pages.

One way we could maybe do this would be through some sort of config that let us specify a list of classes that should be extracted to a separated file, after all the classes needed have been added to the build.

This would leave us with a critical file that we can read and inline in our HTML and another non-critical file that we can request through a `<link>` tag.

And since the process happens after the normal build, the classes added to the non-critical file will have no problem related to the order they appear, since they where simply loaded a little later by the browser. Anything they need for context or specificity would have been added earlier by the critical inlined classes.
